taxonomy/app/taxonomy/services.py

```diff
@@ -163,8 +194,11 @@ def batch_latest_taxonomies(
 def bulk_delete_category_association(
     session: Session,
     category_id: str,
+    tenant: str,
 ) -> None:
-    session.query(AssociationTaxonomyCategory).filter(
-        AssociationTaxonomyCategory.category_id == category_id,
-    ).delete(synchronize_session=False)
+    association = session.query(AssociationTaxonomyCategory).filter(
+        AssociationTaxonomyCategory.category_id == category_id)
+    if association.taxonomy.tenant != tenant:
+        raise CheckFieldError('Wrong tenant')
```

Comment:
Here and above can we provide a more meaningful error message? Something like "Taxonomy is not associated with current tenant"